### PR TITLE
prosody: update to 0.12.4.

### PR DESCRIPTION
**Pull request by AnInternetTroll**
- **New package: lua-unbound-1.0.0**
- **prosody: add lua51-unbound as a dependency**

#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86\_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl

Crossbuilt from x86\_64-glibc and tested on aarch64-musl

Fixes the following warning that shows up when using prosodyctl (any
command):

```
**************************
Prosody was unable to find lua-unbound
This package can be obtained in the following ways:

  Debian/Ubuntu | sudo apt install lua-unbound
       luarocks | luarocks install luaunbound
         Source | https://www.zash.se/luaunbound.html

Old DNS resolver library will be used
More help can be found on our website, at https://prosody.im/doc/depends
**************************
```

Fixes the following deprecation as well:

| Option    | Default | Notes                                                                                                |
|-----------|---------|------------------------------------------------------------------------------------------------------|
| daemonize | true    | Whether to daemonize Prosody or not. DEPRECATED since 0.12, use command line flags -D or -F instead. |

### CODE REVIEW

File: srcpkgs/lua54-unbound/template [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	vinstall lunbound.so 755 usr/lib/lua/5.1/
+
+	cd ${wrksrc}/lua52
+	vinstall lunbound.so 755 usr/lib/lua/5.2/
+
+	cd ${wrksrc}/lua53
+	vinstall lunbound.so 755 usr/lib/lua/5.3/
+
+	cd ${wrksrc}/lua54
+	vinstall lunbound.so 755 usr/lib/lua/5.4/
```
Comment (classabbyamp) [inline srcpkgs/lua54-unbound/template:53]: why not loop these?

File: srcpkgs/lua54-unbound/update [hunk 2]
```diff
@@ -0,0 +1,2 @@
+site="https://code.zash.se/luaunbound/tags"
+pattern='<a href="/luaunbound/rev/\K[\d.]+(?=">)'
```
Comment (classabbyamp) [inline srcpkgs/lua54-unbound/update:2]:
```suggestion
pkgname=luaunbound
```

File: srcpkgs/lua54-unbound/template [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	mkdir -p lua51
+	mv * lua51 || true
+	cp -a lua51 lua52
+	cp -a lua51 lua53
+	cp -a lua51 lua54
+}
+
+do_build() {
+	for _lua_version in $_lua_versions
+	do
```
Comment (classabbyamp) [inline srcpkgs/lua54-unbound/template:32]:
```suggestion
	for _lua_version in $_lua_versions; do
```

File: srcpkgs/lua51-luasocket/template [hunk 4]
```diff
@@ -1,30 +1,97 @@
...
+	cp -a lua51 lua53
+	cp -a lua51 lua54
+}
+
 do_build() {
-	make CC=$CC LD=$CC LUAINC=${XBPS_CROSS_BASE}/usr/include/lua5.1 \
-		PLAT=linux LUAVER=5.1 \
-		${makejobs} all-unix
+	for _lua_version in $_lua_versions
+	do
```
Comment (classabbyamp) [inline srcpkgs/lua51-luasocket/template:31]:
```suggestion
	for _lua_version in $_lua_versions; do
```

File: srcpkgs/lua51-luasocket/template [hunk 5]
```diff
@@ -1,30 +1,97 @@
...
+			LUAV="${_lua_version#lua}" \
+			PLAT=linux \
+			${makejobs} \
+			all-unix
+	done
+}
+
+do_install() {
+	for _lua_version in $_lua_versions
+	do
```
Comment (classabbyamp) [inline srcpkgs/lua51-luasocket/template:47]:
```suggestion
	for _lua_version in $_lua_versions; do
```

**Comment by AnInternetTroll** [comment]
Thanks, completely forgot about those for loops!

**Pull request merged**